services-custom/iam-policy-builder/src/main/java/software/amazon/awssdk/policybuilder/iam/internal/DefaultIamPolicyReader.java

```diff
@@ -132,9 +132,10 @@ private List<IamPrincipal> readPrincipals(Map<String, JsonNode> statementObject,
 
         if (principalsNode.isObject()) {
             List<IamPrincipal> result = new ArrayList<>();
-            principalsNode.asObject().forEach((id, value) -> {
-                result.add(IamPrincipal.create(id, expectString(value, name + " entry value")));
-            });
+            Map<String, JsonNode> principalsNodeObject = principalsNode.asObject();
+            principalsNodeObject.keySet().forEach(
+                k -> result.addAll(IamPrincipal.createAll(k, readStringArray(principalsNodeObject, k)))
+            );
```

Comment:
Does this still work for non-array principals?

Like

```
"Principal": { "AWS": "123456789012" }
```

Reply:
[readStringArray](https://github.com/aws/aws-sdk-java-v2/blob/master/services-custom/iam-policy-builder/src/main/java/software/amazon/awssdk/policybuilder/iam/internal/DefaultIamPolicyReader.java#L180) method is capable of reading non-array principals as well.

Reply:
Cool (if not confusing)! @cenedhryn Can we rename this method while we're here? Maybe read**As**StringArray?

Reply:
I went super literal!